analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -83,6 +83,9 @@ private ProgramState()
         public ProgramState SetException(ExceptionState exception) =>
             this with { Exception = exception };
 
+        public ProgramState ResetException() =>
```

Comment:
This is between two `SetException` overloads, you can move it after the next one.